### DISCUSSION TITLE
Can the values argument in tar_map be a target?

### DISCUSSION THREAD

**Discussion by n8layman** [ropensci/targets#948] [Help]
I'm trying to first create a small target holding a tibble of model names and function calls and then use tar_map's static branching to build all the models as separate targets. I like static branching here because the generated target names make sense and there are not a lot of models. However when I try to call tar_map using a target as the values argument such as

```
tar_map(
    values = model_calls,
    ...
)
```

I get `object 'model_calls' not found` returned by `targets::tar_assert_list(values)`. I can always use `values = tar_read(model_calls)` but that seems to break dependency. So sometimes with that approach the tar_map targets can build before the model_call target. Is there another approach that would let me do this?

modifying the example [here][1] for a more full example, can I do the following or something with the same effect?

```
library(targets)
library(tarchetypes)
library(dplyr)

model_analyses <- tar_plan(

    tar_target(params, tribble(
      ~name, ~arg1,  ~arg2,  
      "A",  "a",  1,    
      "B",  "b",  2,
      "C",  "c",  3  
    )),

    tar_map(
      values = params, 
      names = name,
      tar_target(first_target,
                 f(arg1, arg2)),
      tar_target(second_target,
                 g(first_target)
      )
) 

list(model_analysis)

```

  [1]: https://stackoverflow.com/questions/71274462/targetstar-map-example-r-targets-static-branching

**Comment by wlandau** [comment]
Static branching happens prior to the pipeline, so it assumes no targets have run yet. I recommend that you create the data frame of model calls as a pre-existing global object. You can then reference that and return it as a target if you really want (`tar_target(model_calls_target, model_calls)`).
